### PR TITLE
Add Bytes setting type

### PR DESCRIPTION
**Pull request by jordansissel**
This will be used with persistent queue and eventually can be used for other things.

### CODE REVIEW

File: logstash-core/lib/logstash/settings.rb.orig [hunk 1]
```diff
@@ -0,0 +1,380 @@
+# encoding: utf-8
```
Comment (suyograo) [inline logstash-core/lib/logstash/settings.rb.orig]: Remove this file?

Comment (jordansissel) [inline logstash-core/lib/logstash/settings.rb.orig]: Done. Leftover from some git surgery that went nearly-wrong.

**Comment by jordansissel** [comment]
Rebased on master. Waiting for Travis ....

**Comment by suyograo** [comment]
@jordansissel this is good to merge. Kafka test failures are unrelated to this PR.

**Comment by jordansissel** [comment]
Test failed due to kafka, not this PR.

**Comment by jordansissel** [comment]
Will merge once @colinsurprenant is done suffering through some git merge work. <3

**Comment by jordansissel** [comment]
Looks like this landed along with the PQ merge #6152 and #6163

**Pull request closed without merging**